src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4949,6 +4950,71 @@ public override IOperation VisitMemberInitializer(IMemberInitializerOperation op
             throw ExceptionUtilities.Unreachable;
         }
 
+        public override IOperation VisitAnonymousObjectCreation(IAnonymousObjectCreationOperation operation, int? captureIdForResult)
+        {
+            ImplicitInstanceInfo savedCurrentImplicitInstance = _currentImplicitInstance;
+            _currentImplicitInstance = new ImplicitInstanceInfo((INamedTypeSymbol)operation.Type);
+
+            var properties = operation.Type.GetMembers().OfType<IPropertySymbol>().ToImmutableArray();
+            Debug.Assert(properties.Length == operation.Initializers.Length);
+
+            SpillEvalStack();
```

Comment:
Is it possible to have an anonymous creation without initializers? In that case, should we not spill the eval stack?

Reply:
It is allowed in C#, generates `BC36574: Anonymous type must contain at least one member.` in VB. I have added an early bailout for no initializers and also added unit tests.

---
In reply to: [192900607](https://github.com/dotnet/roslyn/pull/27370#discussion_r192900607) [](ancestors = 192900607)